src/org/camunda/latera/bss/connectors/hoper/hydra/File.groovy

```diff
@@ -16,6 +16,10 @@ trait File {
     return getFileEntityType(getSubjectEntityType(subjectId), id)
   }
 
+  Map getContractFileEntityType(def contractId, def id = null) {
+    return getFileEntityType(getContractEntityType(contractId), id)
```

Comment:
No, this is not really how it works.
Firstly, [getContractEntityType](https://github.com/latera/camunda-ext/blob/master/src/org/camunda/latera/bss/connectors/hoper/hydra/Contract.groovy#L14) requires not only contract id itself, but also parent entity url part because for contract creation/update/delete urls should looks like `subjects/customers/:customer_id/contracts/:contract_id`. You cannot just skip parent part here because url will be malformed. Also calling this method without passing a parent will cause exception.

Secondly, there is no entpoint like `subjects/customers/:customer_id/contracts/:contract_id/files` you're trying to use for attaching files to contract. Real url should look like `documents/:document:id/files`.

Thirdly, you can attach files to any document as well as any subject. Currently there are methods to work with any subject files like getSubjectFiles. So here should be getDocumentFiles instead of getContractFiles.